tests/examples/test_examples.py

```diff
@@ -28,8 +48,15 @@
     (os.path.join('tensorflow', 'tf1'), ['-P', 'steps=10']),
     ('xgboost', ['-P', 'learning_rate=0.3', '-P', 'colsample_bytree=0.8', '-P', 'subsample=0.9'])
 ])
-def test_mlflow_run_example(directory, params):
-    cli_run_list = [os.path.join(EXAMPLES_DIR, directory)] + params
+def test_mlflow_run_example(directory, params, tmpdir):
+    example_dir = os.path.join(EXAMPLES_DIR, directory)
+    tmpdir = os.path.join(tmpdir.strpath, directory)
+
+    shutil.copytree(example_dir, tmpdir)
+    conda_yml_path = find_conda_yaml(tmpdir)
+    replace_mlflow_with_dev_version(conda_yml_path, os.path.abspath('.'))
+
+    cli_run_list = [tmpdir] + params
     invoke_cli_runner(cli.run, cli_run_list)
```

Comment:
These lines do:
1. Create a temporary directory (`tmpdir`)
2. Copy contents in the example directory to `tmpdir`
3. Modify `conda.yml` in `tmpdir`
4. Run `mlflow run <path_to_tmpdir>`